core/src/main/scala/akka/persistence/cassandra/query/EventsByTagPublisher.scala

```diff
@@ -260,6 +261,21 @@ private[query] class EventsByTagPublisher(
         stopIfDone()
       else
         buf :+= env
+
+      if (buf.size > maxBufferSize) {
+        log.warning(
+          s"${backtrackingLog}, buffer size exceeded. Shouldn't happen, please report issue. " +
+          s"Got event for tag [{}] from pid [{}] with seqNr [{}] and offset [{}], buffered [${buf.size}]",
+          tag, env.persistentRepr.persistenceId, env.persistentRepr.sequenceNr, formatOffset(env.offset)
+        )
```

Comment:
Yeah, seems a bit dangerous to have a bounds but continue with just a warning if it is exceeded.